test/fetch-mocked.test.js

```diff
@@ -136,20 +155,173 @@ describe('unit tests on fetch functions', function() {
+        done();
+      }, results);
+    });
+
+    it('should return empty list for users with no collections', function(done) {
+      results.userInfo = fixtures.USER_INFO_LISTDB_ONLY;
+
+      fetch.getAllowedCollections(function(err, res) {
+        assert.equal(err, null);
+        assert.deepEqual(res, []);
+        done();
+      }, results);
+    });
   });
+
   describe('getDatabaseCollections', function() {
-    it('should ignore auth errors gracefully');
-    it('should pass on other errors from the listCollections command');
-    // ...
+    var results = {};
+    it('should ignore auth errors gracefully', function(done) {
+      results.db = makeMockDB(new Error('not authorized on fooBarDatabase to execute command '
+        + '{listCollections: true, filter: {}, cursor: {}'), null);
+
+      fetch.getDatabaseCollections(results.db.admin(), function(err, res) {
+        assert.equal(err, null);
+        assert.deepEqual(res, []);
+        done();
+      });
+    });
+
+    it('should pass on other errors from the listCollections command', function(done) {
+      results.db = makeMockDB(new Error('some other error from list collections'), null);
+
+      fetch.getDatabaseCollections(results.db.admin(), function(err, res) {
+        assert.ok(err);
+        assert.equal(err.command, 'listCollections');
+        assert.deepEqual(res, null);
+        done();
+      });
+    });
   });
+
   describe('listCollections', function() {
-    it('should merge the two collection lists correctly');
-    // ...
+    var results = {};
+
+    it('should lists all collections for each listable db', function(done) {
+      results.userInfo = fixtures.USER_INFO_JOHN;
+      results.databases = [
+        {
+          'name': 'accounts'
+        },
+        {
+          'name': 'products'
+        },
+        {
+          'name': 'reporting'
+        },
+        {
+          'name': 'sales'
+        }
+      ];
+      results.db = makeMockDB(null, [{
+        'name': 'testCol'
+      }]);
+
+      fetch.listCollections(function(err, res) {
+        assert.equal(err, null);
+        res.sort();
+        var expected = [
+          {
+            '_id': 'accounts.testCol',
+            'database': 'accounts',
+            'name': 'testCol'
+          },
+          {
+            '_id': 'products.testCol',
+            'database': 'products',
+            'name': 'testCol'
+          },
+          {
+            '_id': 'reporting.testCol',
+            'database': 'reporting',
+            'name': 'testCol'
+          },
+          {
+            '_id': 'sales.testCol',
+            'database': 'sales',
+            'name': 'testCol'
+          }
+        ];
+        expected.sort();
+        assert.deepEqual(res, expected);
+        done();
+      }, results);
+    });
+
+    it('should be empty for no privileges', function(done) {
+      results.userInfo = fixtures.USER_INFO_LISTDB_ONLY;
+      results.databases = [
+        {
```

Comment:
Could be moved to a `beforeEach` function together with the same block above, to make it DRY.